ingestion-beam/src/main/java/com/mozilla/telemetry/contextualservices/ParseReportingUrl.java

```diff
@@ -165,25 +206,28 @@ public Result<PCollection<SponsoredInteraction>, PubsubMessage> expand(
                 .setPosition(parsePosition(event.path("extra")).orElse("no_position"));
           }
 
+          interactionBuilder.setScenario(parseScenario(metrics).orElse(null));
+
           // Set the source based on the value of the docType
-          if (DT_TOPSITES_CLICK.equals(docType) || DT_TOPSITES_IMPRESSION.equals(docType)) {
+          if (DT_TOPSITES.equals(docType) || DT_TOPSITES_CLICK.equals(docType)
+              || DT_TOPSITES_IMPRESSION.equals(docType)) {
             interactionBuilder.setSource(SponsoredInteraction.SOURCE_TOPSITES);
-          } else if (DT_QUICKSUGGEST_IMPRESSION.equals(docType)
+          } else if (DT_QUICKSUGGEST.equals(docType) || DT_QUICKSUGGEST_IMPRESSION.equals(docType)
               || DT_QUICKSUGGEST_CLICK.equals(docType)) {
             interactionBuilder.setSource(SponsoredInteraction.SOURCE_SUGGEST);
```

Comment:
the FoG pings have a `source` metric, but it doesn't seem to be the same as this